### ISSUE TITLE
Export config parameter "maxconnections" seems to apply globally, not per export

### ISSUE THREAD

**Issue by jamespharaoh** [NetworkBlockDevice/nbd#41]
From `man -s5 nbd-server`:

```
       maxconnections
              Optional; integer

              If specified, then it limits the number of opened connections for this export.
```

In real world usage, in the version supplied by ubuntu xenial (16.04), I observe that the maximum connections are enforced as a total for all exports, rather than per-export, as would be expected for an export-level option, and as documented here.

**Comment by yoe** [comment]
Checking the code reveals:

```C
    if (client->server->max_connections > 0 &&
       g_hash_table_size(children) >= client->server->max_connections) {
            msg(LOG_ERR, "Max connections (%d) reached",
                client->server->max_connections);
            goto handler_err;
    }
```

which means that it checks it per-export, but the number of connections counted is global. That is, if you have two exports, where one has a maximum number of connections but the other does not, you'll be able to connect to the one without the maximum number, even while you can't connect to the other anymore.

That is, indeed, obviously wrong. The problem, however, is that negotiation is done in the child process, so fixing this bug is going to take some creative thinking.

**Comment by yoe** [comment]
Fixed by 7e90161
